pkg/asset/manifests/openshift.go

```diff
@@ -81,13 +81,22 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 	platform := installConfig.Config.Platform.Name()
 	switch platform {
 	case awstypes.Name:
-		ssn := session.Must(session.NewSessionWithOptions(session.Options{
-			SharedConfigState: session.SharedConfigEnable,
-		}))
+		ssn, err := installConfig.AWS.Session(context.TODO())
+		if err != nil {
+			return err
+		}
 		creds, err := ssn.Config.Credentials.Get()
 		if err != nil {
 			return err
 		}
+		if !installconfigaws.IsStaticCredentials(creds) {
```

Comment:
I think these checks should be part of the platform validation. They should yield field errors saying that the credentialsMode field is not valid rather than errors saying that the credentials are not valid.

https://github.com/openshift/installer/blob/cdf74784ee90e21f0961fa379e4917c895fafd62/pkg/asset/installconfig/aws/validation.go#L55

Reply:
the requirement for a certain credential source is not mandated by the credentialMode but the aws secret manifest, and therefore i think it's best to check in the credentials manifests asset that i can't move forward with this configuration.
And i disagree the invalid input is credentialMode rather it is the credential source.

Reply:
It is two sides of the same coin.
1) Given that the creds are not static, then the cred mode must be manual.
1) Given that the cred mode is not manual, then the creds must be static.
So, I suppose the question is, given a misconfiguration, which is the user likely to change--the cred mode or the creds? If your contention is that the user is more likely to switch to different creds, then it makes sense to point them at the creds as being invalid. If the user is more likely to switch the cred mode, however, I think it would be better to point the user at the field in the install config that they need to change.

Reply:
> It is two sides of the same coin.
> 
>     1. Given that the creds are not static, then the cred mode must be manual.
> 
>     2. Given that the cred mode is not manual, then the creds must be static.
> 
> So, I suppose the question is, given a misconfiguration, which is the user likely to change--the cred mode or the creds? If your contention is that the user is more likely to switch to different creds, then it makes sense to point them at the creds as being invalid. If the user is more likely to switch the cred mode, however, I think it would be better to point the user at the field in the install config that they need to change.

most users are guided better onto the safe path of user static creds and the defaut creds mode. Manual creds mode has dragons all the way down for now. So i rather have users think that they needs to fix their credentials instead of updating the credentialMode to manual.